src/callback_signer.rs

```diff
@@ -24,7 +25,41 @@ pub trait SignerCallback: Send + Sync {
 ///
 /// Uniffi callbacks are only supported as a method in a structure, so this is a workaround
 pub struct CallbackSigner {
-    signer: c2pa::CallbackSigner,
+    signer: Box<dyn Signer + Sync + Send>,
+}
+
+pub struct RemoteSigner {
+    signer_callback : Box<dyn SignerCallback>,
+    alg: SigningAlg,
+    reserve_size: u32,
```

Comment:
reserve_size really should be a `usize`, but usize doesn't map well in uniffi (it's not a supported type). So I picked u32 so at least most of the time with type coercions we should not encounter truncation issues. But it introduces limitations (since it's technically a usize)...